### PR TITLE
ruff fixed (docstring remaining)

### LINKED ISSUE #99: Fix Linting
The ruff linting is out of date and throwing some errors that are unnecessary. There are also lots of places where the code does not pass the linting. This issue is to track fixing the linting for this repo.

### PR DESCRIPTION
**Pull request by rahul-maurya11b**
# Pull Request

## Description

fixes ruff outdated issue and linting issue

Fixes #99

## How Has This Been Tested?

as of now i have added : "D101","D107","D103","D102","D100" in ignore section  

- [ ] Yes

_If your changes affect data processing, have you plotted any changes? i.e. have you done a quick sanity check?_

- [ ] Yes

## Checklist:

- [ ] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by rahul-maurya11b** [comment]
sir this is not final pr i am checking by removing warnings and linting issue as of now i have added docstring in ignore i will update   that

File: .ruff.toml [hunk 1]
```diff
@@ -1,10 +1,10 @@
 # Enable pycodestyle (`E`) and Pyflakes (`F`) codes by default.
-select = ["E", "F", "D", "I"]
-ignore = ["D200","D202","D210","D212","D415","D105",]
+lint.select = ["E", "F", "D", "I"]
+lint.ignore = ["D200","D202","D210","D212","D415","D105","D101","D107","D103","D102","D100"]
```
Comment (jacobbieker) [inline .ruff.toml:3]: I wouldn't just add more to ignore to fix this, what should be done is fix these linting errors.

Comment (rahul-maurya11b) [reply .ruff.toml:3]: yes sir i am adding dockstrings to fix these errors

**Comment by rahul-maurya11b** [comment]
@jacobbieker  sir docstrings added please check.

**Review by jacobbieker — APPROVED** [review]
This is great! Thank you!

**Pull request merged**